### ISSUE TITLE
Rectangular area per notebook that supports webview? 

### ISSUE THREAD

**Issue by claudiaregio** [microsoft/vscode#105040]
As suggested by Kai, in order to resolve the kernel information/variable explorer experience we would need a rectangular area within the notebook canvas that supports the following things:

- Can be invoked by command palette or icon (in notebook toolbar or cell toolbar or both TBD) 
- Configurable to be pinned at the top or bottom of the canvas
- Resizeable 
- For notebook file type only 

For variable explorer we would need to be able to display a table view whose columns consist of "Name", "Type", "Size" and "Value", give users ability to search within it and potentially have users change/edit in line. 

For kernel area we would need to be able to display kernel information, usage, graphs, etc.

**Comment by jrieken** [comment]
fyi - we are working on "webview views" that can be placed into the usual places (sidebar, panel) and that might be a good starting point

**Comment by claudiaregio** [comment]
Yes we already explored that option, but after discussing with Kai he proposed this solution given the potential issues with using panel or sidebar.

**Comment by greazer** [comment]
In particular, we have a need to have a per notebook area to display kernel information or variables as each notebook has it's own runtime state.

**Comment by claudiaregio** [comment]
Is there any update on this item? 

I could provide a mockup with some examples if that helps paint the picture better.

**Comment by rchiodo** [comment]
Would it be possible to have more than one? I feel like we could use this to get the interactive window working with the new API too.

**Comment by DonJayamanne** [comment]
> Meaning one view pinned to the top and another at the bottom.

Woudln't we instead want a code cell pinned to the bottom, instead of pinning a webview. Else we end up with a lot of work around to get monaco working yet again.

**Comment by rchiodo** [comment]
> Woudln't we instead want a code cell pinned to the bottom, instead of pinning a webview. Else we end up with a lot of work around to get monaco working yet again.

Yeah that would be ideal but if this could get us most of the way there, we could start with this.
